scenarios/getting_started/rust/getting_started/src/main.rs

```diff
@@ -0,0 +1,161 @@
+    // Initialize the logger from the environment
+    env_logger::init();
+
+    dotenv().ok();
+    let conn_settings = connection_settings::get_connection_settings(Some(".env")).unwrap();
+    let mqtt_clean_session = conn_settings.mqtt_clean_session;
+    if !mqtt_clean_session {
+        println!("This sample does not support connecting with existing sessions");
+        std::process::exit(1);
+    }
+    let port = conn_settings.mqtt_tcp_port;
+    let hostname = conn_settings.mqtt_host_name;
+    let keepalive = conn_settings.mqtt_keep_alive_in_seconds;
+    let client_id = conn_settings.mqtt_client_id;
+    let mut protocol = "tcp";
+
+    // Set up connection builder
+    let mut conn_opts_builder = mqtt::ConnectOptionsBuilder::with_mqtt_version(MQTT_VERSION_5);
+    conn_opts_builder
+        .keep_alive_interval(Duration::from_secs(keepalive.into()))
+        .clean_start(mqtt_clean_session);
+
+    if let Some(username) = &conn_settings.mqtt_username {
+        conn_opts_builder.user_name(username);
+    }
+
+    if let Some(password) = &conn_settings.mqtt_password {
+        conn_opts_builder.password(password);
+    }
+
+    if conn_settings.mqtt_use_tls {
+        protocol = "mqtts";
+
+        // Create an SSL options builder
+        let mut ssl_opts_builder = SslOptionsBuilder::new();
+        ssl_opts_builder.ssl_version(SslVersion::Tls_1_2);
+
+        // Trust store (CA file)
+        if let Some(ca_file_path) = conn_settings.mqtt_ca_file {
+            // Handle the Result returned by trust_store
+            match ssl_opts_builder.trust_store(ca_file_path) {
+                Ok(_) => {
+                    // Trust store loaded successfully, continue
+                }
+                Err(err) => {
+                    eprintln!("Failed to load trust store: {:?}", err);
+                    return;
+                }
+            }
+        }
+
+        // Certificate file
+        if let Some(cert_file_path) = conn_settings.mqtt_cert_file {
+            // ssl_opts_builder = ssl_opts_builder.key_store(cert_file_path);
+            match ssl_opts_builder.key_store(cert_file_path) {
+                Ok(_) => {
+                    // Key store loaded successfully, continue
+                }
+                Err(err) => {
+                    eprintln!("Failed to load key store: {:?}", err);
+                    return;
+                }
+            }
+        }
+
+        if let Some(key_file_path) = conn_settings.mqtt_key_file {
+            match ssl_opts_builder.private_key(key_file_path) {
+                Ok(_) => {
+                    // Key store loaded successfully, continue
+                }
+                Err(err) => {
+                    eprintln!("Failed to load private key: {:?}", err);
+                    return;
+                }
+            }
+        }
+
+        if let Some(key_password) = conn_settings.mqtt_key_file_password {
+            ssl_opts_builder.private_key_password(key_password);
```

Comment:
have you been able to test certs with key passwords?

Reply:
Will check.

Reply:
tested - it works

Reply:
awesome.. I'd like to learn how did you produce a key with a password, so we can also validate in other languages.